### PR TITLE
Update default SSL version to TLSv1_2 for RUBY_VERSION >= 2.0.0

### LINKED ISSUE #86: Update default `http_ssl_version`
Getting the following error with the latest version of openssl on Ubuntu 20.04:
`OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: no protocols available`

```
apt-cache policy openssl
openssl:
  Installed: 1.1.1f-1ubuntu2
  Candidate: 1.1.1f-1ubuntu2
```

I can get it to work by using `Postmark::ApiClient.new(api_token, {http_ssl_version: :TLSv1_2})`.

Calling `openssl s_client -connect api.postmarkapp.com:443` shows that the certificate uses `TLSv1.2`:

```
New, TLSv1.2, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
```

### PR DESCRIPTION
**Pull request by nickhammond**
To resolve #86.

### CODE REVIEW

**Comment by tomazy** [comment]
@nickhammond what's up with the failing test?

**Comment by nickhammond** [comment]
@tomazy Not sure, triggered a rebuild and it passed. Nothing obvious in the failing spec details:

```
  1) Mail::Postmark#deliver! returns self by default
     Failure/Error: expect(message.deliver).to eq message
       expected: #<Mail::Message:12982, Multipart: false, Headers: <From: sheldon@bigbangtheory.com>, <To: lenard@bigbangtheory.com>, <Subject: Hello!>>
            got: #<Mail::Message:12982, Multipart: false, Headers: <From: sheldon@bigbangtheory.com>, <To: lenard@bigbangtheory.com>, <Subject: Hello!>>
       (compared using ==)
     # ./spec/unit/postmark/handlers/mail_spec.rb:28:in `block in (root)'
```

**Review by tomazy — APPROVED** [review]
👍

File: lib/postmark/http_client.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ class HttpClient
       :path_prefix => '/',
       :http_read_timeout => 15,
       :http_open_timeout => 5,
-      :http_ssl_version => :TLSv1
+      :http_ssl_version => (RUBY_VERSION >= '2.0.0' ? :TLSv1_2 : :TLSv1)
```
Comment (ibalosh) [inline lib/postmark/http_client.rb:22]: I noticed mentions of max_version, min_version options. I see these in Ruby 2.4.0 and above.
Wondering if we should set version in these higher versions this way. 

https://github.com/wildbit/postmark-gem/issues/86

**Comment by rassom** [comment]
@postmark Please merge this PR!

**Comment by nickhammond** [comment]
Closing in favor of #97.

**Pull request closed without merging**